airflow-core/src/airflow/models/dagbundle.py

```diff
@@ -36,7 +36,7 @@ class DagBundleModel(Base):
     """
 
     __tablename__ = "dag_bundle"
-    name = Column(StringID(), primary_key=True)
+    name = Column(StringID(), primary_key=True, nullable=False)
```

Comment:
You need to also migrate this change. I think that's what's causing the error you got. dag_bundle.name is nullable but dag.bundle_name is not nullable so both columns are not compatible to be used as FK